### ISSUE TITLE
Add LoRA training via VERL integration

### ISSUE THREAD

**Issue by ash-ding** [akashgit/remote-factory#1181]
## Description

Currently model weights don't update between iterations — the vLLM backend uses a static model. Need real RL training to improve policy over time.

## Background

**Current workflow:**
1. Lumen Context Agent generates 8 prompts
2. vLLM generates 64 rollouts per prompt (512 total)
3. Evaluator scores solutions
4. **Model stays the same** → next iteration uses same policy

**Desired workflow:**
1. Generate prompts
2. Generate rollouts
3. Evaluate scores
4. **Train LoRA weights** on best rollouts ← NEW STEP
5. Update vLLM with new weights → next iteration uses improved policy

## Proposed Implementation

Create `factory/rl/verl_trainer.py` that integrates with Discover's VERL code:

```python
"""VERL-based LoRA training for Einstein Arena."""
from pathlib import Path

def train_on_rollouts(
    rollouts: list[dict],
    advantages: list[float],
    model_path: Path,
    lora_output_path: Path,
    num_gpus: int = 4,
) -> None:
    """Train LoRA weights using VERL PPO.
    
    Args:
        rollouts: Rollouts with prompt, solution, thinking, code
        advantages: Advantage scores (higher = better)
        model_path: Base model path (e.g., ~/models/Qwen3-32B)
        lora_output_path: Where to save LoRA checkpoint
        num_gpus: GPUs for training (4 for Actor, 4 for Rollout)
    """
    # 1. Convert rollouts to VERL format
    # 2. Compute advantages from scores
    # 3. Launch VERL PPO training
    # 4. Save LoRA checkpoint
```

**Workflow integration:**

Add new `FnNode` after evaluation:

```python
FnNode(
    id="rl_train_lora",
    command="python3 -m factory.rl.verl_trainer ...",
    reads={".factory/rl/iteration_{}/evaluation_results.json"},
    writes={".factory/rl/checkpoints/lora_iter_{}.pt"},
)
```

## Dependencies

**External:**
- Discover codebase at `/workspace/home/asherding/code/discover`
- VERL library (from Discover)
- Ray for distributed training
- PyTorch with FSDP

**Hardware:**
- 8 GPUs recommended (4 for Actor, 4 for Rollout)
- Can run on fewer but slower

## Tasks

- [ ] Create `factory/rl/verl_trainer.py`
- [ ] Implement rollout → VERL format conversion
- [ ] Implement advantage computation from scores
- [ ] Integrate VERL PPO training loop
- [ ] Add checkpoint saving/loading
- [ ] Update Einstein Arena workflow to call training step
- [ ] Add `--skip-training` flag for debugging
- [ ] Document GPU requirements

## Challenges

1. **Environment setup** — VERL has complex dependencies
2. **Distributed training** — Need Ray + FSDP config
3. **Hyperparameters** — Learning rate, batch size, PPO epochs, etc.
4. **Convergence** — May need many iterations to see improvement

## Acceptance Criteria

- [ ] Model policy improves over iterations (scores trend upward)
- [ ] LoRA weights saved to `.factory/rl/checkpoints/lora_iter_N.pt`
- [ ] Can resume from checkpoint
- [ ] Training completes in reasonable time (<1 hour per iteration)
- [ ] Works with 4-8 GPUs

## Estimate

**1-2 weeks** — complex integration with external codebase

## Priority

**Low** — Nice-to-have, not required for MVP v2. Current workflow with static model still valuable for:
- Testing infrastructure
- Evaluating prompts
- Benchmarking different base models

## Related

- Discover codebase: `/workspace/home/asherding/code/discover`
- VERL paper: https://arxiv.org/abs/2409.03005
- Issue #1176 (vLLM backend) — should complete first

**Comment by ash-ding** [comment]
Wrong repo - should be in fork ash-ding/remote-factory
